### PR TITLE
[new release] devkit (1.20240429)

### PR DESCRIPTION
**Pull request by Khady**
Development kit - general purpose library

- Project page: <a href="https://github.com/ahrefs/devkit">https://github.com/ahrefs/devkit</a>

##### CHANGES:

- add opentelemetry support
- remove depencency on ocamlnet

### CODE REVIEW

File: packages/devkit/devkit.1.20240429/opam [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+homepage: "https://github.com/ahrefs/devkit"
+dev-repo: "git+https://github.com/ahrefs/devkit.git"
+bug-reports: "https://github.com/ahrefs/devkit/issues"
+synopsis: "Development kit - general purpose library"
+build: [
+  ["dune" "subst"] {dev}
+  ["dune" "build" "-p" name "-j" jobs "@install" "@runtest"{with-test} "@doc"{with-doc}]
+]
+depends: [
+  "ocaml" {>= "4.05.0"}
```
Comment (mseri) [inline packages/devkit/devkit.1.20240429/opam:14]:
```suggestion
  "ocaml" {>= "4.13.0"}
```
Due to

[`failed: Unbound value StringLabels.starts_with`](https://opam.ci.ocaml.org/github/ocaml/opam-repository/commit/fc5a99f1d19925c70c4559cbe5eb2b24d2d1a9bf/variant/compilers,4.12,devkit.1.20240429)

**Comment by mseri** [comment]
All tests are failing with linking errors

**Comment by raphael-proust** [comment]
The tests with OCaml5+ are indeed failing. The test executable doesn't link:

```
- File "dune", line 69, characters 8-12:
- 69 |   (name test)
-              ^^^^
- (cd _build/default && /home/opam/.opam/5.0/bin/ocamlopt.opt -w -40 -g -o test.exe /home/opam/.opam/5.0/lib/ocaml/unix/unix.cmxa /home/opam/.opam/5.0/lib/ocaml/threads/threads.cmxa /home/opam/.opam/5.0/lib/curl/curl.cmxa -I /home/opam/.opam/5.0/lib/curl /home/opam/.opam/5.0/lib/lwt/lwt.cmxa /home/opam/.opam/5.0/lib/ocplib-endian/ocplib_endian.cmxa /home/opam/.opam/5.0/lib/ocplib-endian/bigstring/ocplib_endian_bigstring.cmxa /home/opam/.opam/5.0/lib/lwt/unix/lwt_unix.cmxa -I /home/opam/.opam/5.0/lib/lwt/unix /home/opam/.opam/5.0/lib/curl/curl_lwt.cmx -I /home/opam/.opam/5.0/lib/curl /home/opam/.opam/5.0/lib/extlib/extlib.cmxa /home/opam/.opam/5.0/lib/extunix/ExtUnix.cmxa -I /home/opam/.opam/5.0/lib/extunix /home/opam/.opam/5.0/lib/libevent/liboevent.cmxa -I /home/opam/.opam/5.0/lib/libevent /home/opam/.opam/5.0/lib/pcre/pcre.cmxa -I /home/opam/.opam/5.0/lib/pcre /home/opam/.opam/5.0/lib/ocaml/str/str.cmxa ocamlnet_lite/ocamlnet_lite.cmxa /home/opam/.opam/5.0/lib/stdlib-shims/stdlib_shims.cmxa /home/opam/.opam/5.0/lib/trace/core/trace_core.cmxa /home/opam/.opam/5.0/lib/yojson/yojson.cmxa /home/opam/.opam/5.0/lib/zip/zip.cmxa -I /home/opam/.opam/5.0/lib/zip devkit_core.cmxa devkit.cmxa /home/opam/.opam/5.0/lib/ounit2/advanced/oUnitAdvanced.cmxa /home/opam/.opam/5.0/lib/ounit2/oUnit.cmxa .test.eobjs/native/dune__exe.cmx .test.eobjs/native/dune__exe__Test_httpev.cmx .test.eobjs/native/dune__exe__Test.cmx)
- /usr/bin/ld: /home/opam/.opam/5.0/lib/libevent/libmloevent.a(event_stubs.o): in function `event_cb':
- event_stubs.c:(.text+0xd1): undefined reference to `callback3'
- /usr/bin/ld: /home/opam/.opam/5.0/lib/libevent/libmloevent.a(event_stubs.o): in function `oc_event_base_init':
- event_stubs.c:(.text+0x9b0): undefined reference to `invalid_argument'
- collect2: error: ld returned 1 exit status
- File "caml_startup", line 1:
- Error: Error during linking (exit code 1)
[ERROR] The compilation of devkit.1.20240429 failed at "dune build -p devkit -j 255 @install @runtest".
```

**Comment by Khady** [comment]
The error was actually because of `libevent.0.8.1`. I've rebased the branch, the CI is now installing libevent.0.9.0 which was just merged, and the tests now are passing.

File: packages/devkit/devkit.1.20240429/opam [hunk 2]
```diff
@@ -0,0 +1,51 @@
+opam-version: "2.0"
+maintainer: "igor@ahrefs.com"
+authors: "Ahrefs <github@ahrefs.com>"
+license: "LGPL-2.1-only with OCaml-LGPL-linking-exception"
```
Comment (avsm) [inline packages/devkit/devkit.1.20240429/opam:4]:
```suggestion
license: "LGPL-2.1-only WITH OCaml-LGPL-linking-exception"
```

fixing linting error

**Comment by mseri** [comment]
The remaining failure seems to be a problem in libevent.0.9.0 with macos arm:
```
#=== ERROR while compiling libevent.0.9.0 =====================================#
# context              2.2.0~beta1 | macos/arm64 | ocaml-base-compiler.4.14.2 | file:///Users/mac1000/opam-repository
# path                 ~/.opam/4.14.2/.opam-switch/build/libevent.0.9.0
# command              ~/.opam/opam-init/hooks/sandbox.sh build make depend
# exit-code            2
# env-file             ~/.opam/log/libevent-39041-0cb22f.env
# output-file          ~/.opam/log/libevent-39041-0cb22f.out
### output ###
# Makefile:99: depend: No such file or directory
# gcc -I /Users/mac1000/.opam/4.14.2/lib/ocaml -MM *.c > depend
# event_stubs.c:13:10: fatal error: 'event.h' file not found
# #include <event.h>
#          ^~~~~~~~~
# 1 error generated.
# make: *** [depend] Error 1

```

**Pull request merged**